apps/backend/src/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,244 @@
+      jest.spyOn(userService, 'create').mockResolvedValueOnce(userMock);
+      await expect(service.signUp(userMock)).resolves.toEqual(userMock);
+    });
+  });
+
+  describe('signIn', () => {
+    it('should generate a pair of tokens if user successfully logged', async () => {
+      const user = {
+        id: 'someId',
+        ...userMock,
+        password: await hash(userMock.password, SALT_ROUNDS),
+      };
+      jest.spyOn(userService, 'findByEmail').mockResolvedValueOnce(user);
+      const spiedFunction = jest
+        .spyOn(userService, 'update')
+        .mockResolvedValueOnce(null);
+
+      const tokens = await service.signIn({
+        email: userMock.email,
+        password: userMock.password,
+      });
+
+      expect(tokens).toEqual(
+        expect.objectContaining({
+          accessToken: expect.any(String),
+          refreshToken: expect.any(String),
+        }),
+      );
+      expect(spiedFunction).toBeCalled();
+    });
+
+    it('should throw if user not found', async () => {
+      jest.spyOn(userService, 'findByEmail').mockResolvedValueOnce(null);
+      await expect(
+        service.signIn({
+          email: userMock.email,
+          password: userMock.password,
+        }),
+      ).rejects.toThrowError(BadRequestException);
+    });
+
+    it('should throw if password do not match', async () => {
+      const user = {
+        ...userMock,
+        password: await hash(userMock.password + 'diff_password', SALT_ROUNDS),
+      };
+
+      jest.spyOn(userService, 'findByEmail').mockResolvedValueOnce(user);
+
+      await expect(
+        service.signIn({
+          email: userMock.email,
+          password: userMock.password,
+        }),
+      ).rejects.toThrowError(BadRequestException);
+    });
+  });
+
+  describe('logout', () => {
+    it('should update users refresh token to null', async () => {
+      const spiedFunction = jest.spyOn(userService, 'update');
+
+      await service.logout('someId');
+
+      expect(spiedFunction).toBeCalledWith('someId', {
+        refreshToken: null,
+      });
+    });
+  });
+
+  describe('refreshTokens', () => {
+    let user;
+    let tokens;
+
+    beforeEach(async () => {
+      user = {
+        id: 'someId',
+        ...userMock,
+        password: await hash(userMock.password, SALT_ROUNDS),
+      };
+
+      jest.spyOn(userService, 'findByEmail').mockResolvedValueOnce(user);
+      jest.spyOn(userService, 'update').mockResolvedValueOnce(null);
+
+      tokens = await service.signIn({
+        email: userMock.email,
+        password: userMock.password,
+      });
+    });
+
+    it('should generate new pair of tokens', async () => {
+      jest.spyOn(userService, 'findById').mockResolvedValueOnce({
+        ...user,
+        refreshToken: await argon.hash(tokens.refreshToken),
+      });
+
+      // I need to wait a second because if I don't the tokens will be the same
+      await new Promise((r) => setTimeout(r, 1000));
```

Comment:
Try to mock the module responsible for making you await as we talked about on discord.